packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -91,17 +132,84 @@ class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProv
     return ScrollbarPainter(
       color: _kScrollbarColor,
       textDirection: _textDirection,
-      thickness: _kScrollbarThickness,
+      thickness: _thickness,
       fadeoutOpacityAnimation: _fadeoutOpacityAnimation,
       mainAxisMargin: _kScrollbarMainAxisMargin,
       crossAxisMargin: _kScrollbarCrossAxisMargin,
-      radius: _kScrollbarRadius,
+      radius: _radius,
       padding: MediaQuery.of(context).padding,
       minLength: _kScrollbarMinLength,
       minOverscrollLength: _kScrollbarMinOverscrollLength,
     );
   }
 
+  // Handle a gesture that drags the scrollbar by the given amount.
+  void _dragScrollbar(double primaryDelta) {
+    assert(widget.controller != null);
+
+    // Convert primaryDelta, the amount that the scrollbar moved, into the
+    // coordinate space of the scroll position, and scroll to that
+    // position.
+    final double viewHeight = widget.controller.position.viewportDimension;
```

Comment:
The scrollbar's track extended is usually less than or equal to `widget.controller.position.viewportDimension` iirc. For example `CupertinoScrollbar` has a fixed mainAxisMargin as well as a mainAxisPadding introduced by mediaQuery (line 140 in this file).
```dart
final double scrollbarExtent = viewHeight - mainAxisMargin - (widget.controller.position.axisDirection == ..... ? padding.horizontal : padding.vertical);
```

Reply:
Good point, I reproduce cases where the thumb drifts because of this.  I'm working on a fix.

Reply:
I fixed this by reusing the logic that's already in ScrollbarPainter, so the two calculations should always match now.